### PR TITLE
Enhance PatternFilter and add unit tests for analyzers

### PR DESCRIPTION
**Pull request by eliosf27**
- Updated PatternFilter to check for attribute existence before accessing it, preventing potential errors.
- Added error handling for invalid regex patterns in the _apply_text_filters method, raising a DataspotError with a descriptive message.
- Introduced comprehensive unit tests for the Analyzer, Base, Compare, Discovery, Finder, Preprocessor, and Tree classes, covering initialization, execution, and edge cases to ensure robust functionality and error handling.

### CODE REVIEW

File: dataspot/analyzers/filters.py [hunk 1]
```diff
@@ -78,9 +79,13 @@ def _apply_numeric_filter(
...
         """
         if operator == ">=":
-            return [p for p in patterns if getattr(p, attr) >= value]
+            return [
+                p for p in patterns if hasattr(p, attr) and getattr(p, attr) >= value
+            ]
         else:  # operator == "<="
-            return [p for p in patterns if getattr(p, attr) <= value]
+            return [
+                p for p in patterns if hasattr(p, attr) and getattr(p, attr) <= value
```
Comment (eliosf27) [inline dataspot/analyzers/filters.py:87]: @claude, what do you think about this change?
